packages/itwinui-variables/src/themes/light.scss

```diff
@@ -143,7 +143,7 @@ $opacities: (
   --iui-color-text-hover: #{hex($fg-100)};
   --iui-color-text-muted: #{hex($fg-500)};
   --iui-color-text-placeholder: #{hex($fg-650)};
-  --iui-color-text-disabled: #{hex($fg-700)};
+  --iui-color-text-disabled: #{hex($fg-900)};
```

Comment:
Last week I had a thought: what if we decrease contrast for disabled items during resting state but we make it 4.5 when the disabled item is hovered or focused?

This would prevent an undesirable situation where the user is focusing an item but can't quite read the text inside it.

Reply:
We can try it.  I do wonder if that visual change will confuse a user to think the element is actionable / clickable.

Reply:
We'd need to introduce a new CSS variable, `--iui-color-text-disabled-hover`, right?

Reply:
Introduced `--iui-color-text-disabled-hover` & `--iui-color-icon-disabled-hover`, implemented only on `.iui-field` thus far.

Reply:
We should do the same in other interactive components. MenuItem is a must, since that's where the original issue was reported.

Reply:
Had to fix #2192 first.  Since that merged, I've added `:hover` & `:focus-visible` to a list item that is disabled.  I did _not_ include `[data-iui-actionable='true']` because I am assuming if a list item is disabled it must also be actionable?  Not sure if that's a safe assumption or not.

Reply:
It's not a safe assumption. `ListItem` is a low-level primitive so it can be used in any arbitrary ways. The `:hover` styling should only apply when it's actionable.

Reply:
Ok, more specifically targeting `[data-iui-actionable='true']` now.

Reply:
Should it also be targeting `:has(.iui-link-action)`?

Reply:
Try checking again, I increased the difference between disabled and disabled hover in dark theme.  The disabled hover state was not very noticeable only in dark mode.

Reply:
Looks good now 👍

Could you also update the PR description and mention the new contrast ratios?

Reply:
Updated the changesets and PR description, will add %s in the coming minutes.

Reply:
While testing the contrast scores I noticed 2 things:
1. I was mainly testing against `iui-color-background-disabled` which gives the lowest score for light themes, but `iui-color-background` gives a lower score for dark themes.  Because of this our dark high contrast theme's disabled hover is not meeting 7.5.
2. Dark high contrast's muted text is scoring 7.48, 0.02 below the required minimum. 🫣

Reply:
It's ok, because:
- [SC 1.4.6](https://www.w3.org/WAI/WCAG22/Understanding/contrast-enhanced.html) only requires 7:1, not 7.5:1. And disabled controls are specifically exempt.
- We are never going to pass all success criteria for AAA (it is near-impossible to do so). There is no such thing as "partial success"; it's either pass or fail. In other words, we can't make claims like "we are meeting AAA color contrast but not the other success criteria".